crates/autopilot/src/domain/fee/mod.rs

```diff
@@ -92,3 +74,89 @@ pub enum Policy {
         factor: f64,
     },
 }
+
+impl Policy {
+    pub fn with(
+        self,
+        order: &boundary::Order,
+        quote: &domain::Quote,
+        skip_market_orders: bool,
+    ) -> Option<Policy> {
+        match order.metadata.class {
+            boundary::OrderClass::Market => {
+                if skip_market_orders {
+                    None
+                } else {
+                    Some(self)
```

Comment:
Since this feature will be released only once the fee=0 feature is live (meaning all the orders sent to us are limit orders, including orders from integrators), I believe we should set `None` in this case also (just in case of some bug). We don't want to charge protocol fee for `Market` order since the driver won't know how to implement custom prices in SettlementEncoder for `Market` orders and we will end up charging fees for orders for which the driver did not keep the funds in the settlement contract (solvers will be losing money).

Reply:
Removed